### ISSUE TITLE
Support `forward` kwargs in `AttributionExplainer` classes

### ISSUE THREAD

**Issue by gsarti** [FOR-sight-ai/interpreto#37]
## Description

Currently `AttributionExplainer` methods only take a single tensor as input for the model forward pass. However, Huggingface-based Transformer models take several arguments as forward inputs (most notably, the `input_ids` and the `attention_mask`, plus potentially other things). Attribution explainers should support an arbitrary number of kwargs to be passed to the model forward.

Example of how this is handled in Captum for Integrated Gradients:

> `additional_forward_args (Any, optional)` – If the forward function requires additional arguments other than the inputs for which attributions should not be computed, this argument can be provided. It must be either a single additional argument of a Tensor or arbitrary (non-tuple) type or a tuple containing multiple additional arguments including tensors or any arbitrary python types. These arguments are provided to forward_func in order following the arguments in inputs. For a tensor, the first dimension of the tensor must correspond to the number of examples. It will be repeated for each of n_steps along the integrated path. For all other types, the given argument is used for all forward evaluations. Note that attributions are not computed with respect to these arguments. Default: None

```python
from transformers import BertTokenizer, BertForMaskedLM
import torch
from captum.attr import LayerIntegratedGradients

model = BertForMaskedLM.from_pretrained("huawei-noah/TinyBERT_General_4L_312D")
tokenizer = BertTokenizer.from_pretrained("huawei-noah/TinyBERT_General_4L_312D")
model.eval()
model.zero_grad()

txt = "Hello [MASK], my name is BERT"
inputs = tokenizer(txt, return_tensors="pt")
input_ids = inputs["input_ids"]
token_type_ids = inputs["token_type_ids"]
attention_mask = inputs["attention_mask"]
ref_input_ids = torch.zeros_like(input_ids)
ref_input_ids[:, 2] = tokenizer.mask_token_id

def predict(input_ids, token_type_ids, attention_mask):
    outputs = model(input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask).logits
    masked_index = torch.nonzero(input_ids == tokenizer.mask_token_id, as_tuple=True)
    out_logits = outputs[masked_index[0], masked_index[1], :]
    # Return max logit value
    return out_logits.max(dim=1).values


lig = LayerIntegratedGradients(predict, model.bert.embeddings)
attributions, delta = lig.attribute(
    inputs=input_ids,
    baselines=ref_input_ids,
    additional_forward_args=(token_type_ids, attention_mask),
    return_convergence_delta=True
)
```

How we handle the creation of the compatible dict in Inseq: https://github.com/inseq-team/inseq/blob/1078e1917834f00139dd369e7eb3e48abfed795c/inseq/models/decoder_only.py#L55

cc @AntoninPoche
